bin/skins/Default/scripts/downloadscreen.lua

```diff
@@ -254,26 +254,68 @@ function archive_callback(entries, id)
     local songsfolder = dlScreen.GetSongsPath()
     res = {}
     folders = { songsfolder .. "/nautica/" }
-    local hasFolder = false
+
+    local longestCommonPath = nil
     for i, entry in ipairs(entries) do
-        for j = 1, #entry do
-            if entry:sub(j,j) == '/' then
-               hasFolder = true
-               table.insert(folders, songsfolder .. "/nautica/" .. entry:sub(1,j))
+        if longestCommonPath == nil then
+            longestCommonPath = entry
+        else
+            if #entry < #longestCommonPath then
+                longestCommonPath = longestCommonPath:sub(1, #entry)
+            end
+            for j=1, #longestCommonPath do
+                if longestCommonPath:sub(j, j) ~= entry:sub(j, j) then
+                    longestCommonPath = longestCommonPath:sub(1, j-1)
+                    break
+                end
+            end
+            if #longestCommonPath == 0 then
+                break
             end
         end
-        game.Log(entry, 0)
-        res[entry] = songsfolder .. "/nautica/" .. entry
     end
-    
-    if not hasFolder then
-        for i, entry in ipairs(entries) do
-            res[entry] = songsfolder .. "/nautica/" .. id .. "/" .. entry
+
+    -- There was nothing!
+    if longestCommonPath == nil then
+        return res
+    end
+
+    for j=#longestCommonPath, 1, -1 do
+        if longestCommonPath:sub(j, j) == '/' then
+            longestCommonPath = longestCommonPath:sub(1, j)
+            break
         end
-        table.insert(folders, songsfolder .. "/nautica/" .. id .. "/")
     end
+
+    game.Log("CommonPath: " .. longestCommonPath, 0)
+
+    local root = songsfolder .. "/nautica/" .. id
```

Comment:
I was planning to modify the chart directory name based on song title, but didn't do it.